[Reference Architecture] Add updates and diagram to CDN Reference Architecture (#9491)
cloudflare/cloudflare-docs: content/reference-architecture/cdn-reference-architecture.md

While Cache Reserve can function without Argo Tiered Cache enabled, it is recommended that Argo Tiered Cache be enabled with Cache Reserve. Argo Tiered Cache will funnel, and potentially eliminate, requests to Cache Reserve which eliminates redundant read operations and redundant storage of cached content reducing egress and storage fees. Enabling Cache Reserve via the Cloudflare dashboard will check and provide a warning if you try to use Cache Reserve without Argo Tiered Cache enabled.

Cache Reserve has a retention period of 30 days which means it will hold cached content for 30 days regardless of cached headers or TTL policy. The TTL policy still affects the content’s freshness which means when content cache TTL expires inside of Cache Reserve, the content will need to be revalidated by checking the origin for any updates. The TTL policy can be set by any number of methods, such as Cache-Control, CDN-Cache-Control response headers, Edge Cache TTL, cache TTL by status code, or Cache Rules. Every time cache is read from Cache Reserve, the retention timer is reset to 30 days. After 30 days, if the cached content has not been read from Cache Reserve, the cache will be deleted.

> There are 3 main criteria to match for content to be considered cacheable via Cache Reserve:

Reviewer: ```suggestion
There are three main criteria to match for content to be considered cacheable via Cache Reserve:
```

> 1. The content must be cacheable. See the [Cache documentation](/cache/) for more details on cacheable conent.

Reviewer: ```suggestion
1. The content must be cacheable. Refer to the [Cache documentation](/cache/) for more details on cacheable content.
```

Author: Already got this one!

2. TTL is set to at least 10 hours. This can be set by any method from the previous paragraph.

> 3. The Content-Length header must be used in the response header. Please note, this means that the [Transfer-Method “chunked”](/support/speed/essentials/why-is-my-dynamic-content-being-sent-with-chunked-encoding/) will prevent Cache Reserve from being populated.

Reviewer: ```suggestion
3. The Content-Length header must be used in the response header. Note that this means the [Transfer-Method “chunked”](/support/speed/essentials/why-is-my-dynamic-content-being-sent-with-chunked-encoding/) will prevent Cache Reserve from being populated.
```

> When combined with Argo Tiered Caching and Argo Smart Routing, Cache Reserve can be a powerful tool for increasing cache hits and in-turn reducing load on origin servers while also improving performance by bringing the content closer to the end user.

Reviewer: I don't think in-turn needs to be hyphenated. The sentence is kinda long, and not that removing in-turn makes it that much shorter, but it's not really adding much either.

Author: Done!

{{<Aside type="note">}}
 

> Using [Image Resizing](/images/image-resizing/) with Cache Reserve will not result in resized images being stored in Cache Reserve since Image Resizing takes place after reading from Cache Reserve. Resized images will be cached in other available tiers when they are served after resizing.

Reviewer: ```suggestion
Using [Image Resizing](/images/image-resizing/) with Cache Reserve will not result in resized images being stored in Cache Reserve because Image Resizing takes place after reading from Cache Reserve. Resized images will be cached in other available tiers when they are served after resizing.
```

 
{{</Aside>}}

### Cache Reserve topology

> Figure 7 illustrates how Cache Reserve can help reduce load on an origin server while also helping repopulate cache stores in both upper and lower tier data centers.

Reviewer: ```suggestion
Figure 7 illustrates how Cache Reserve can help reduce load on an origin server while also repopulating cache stores in both upper and lower tier data centers.
```
